test/test-driver.js

```diff
@@ -12,9 +15,39 @@ exports['test uuid'] = assert => {
   assert.notEqual(/^[a-f0-9-]{36}$/.exec(uuid), null);
 };
 
+exports['test sync device counts'] = promiseTest(assert => Task.spawn(function*() {
+  Preferences.set('services.sync.clients.devices.desktop', 4);
+  Preferences.set('services.sync.clients.devices.mobile', 5);
+  Preferences.set('services.sync.numClients', 9);
+  let client = yield driver.client();
+  assert.equal(client.syncMobileDevices, 5);
+  assert.equal(client.syncDesktopDevices, 4);
+  assert.equal(client.syncTotalDevices, 9);
+
+  Preferences.reset('services.sync.clients.devices.desktop');
+  client = yield driver.client();
+  assert.equal(client.syncMobileDevices, 5);
+  assert.equal(client.syncDesktopDevices, 0);
+  assert.equal(client.syncTotalDevices, 9);
+
+  Preferences.reset('services.sync.clients.devices.mobile');
+  client = yield driver.client();
+  assert.equal(client.syncMobileDevices, 0);
+  assert.equal(client.syncDesktopDevices, 0);
+  assert.equal(client.syncTotalDevices, 9);
+
+  Preferences.reset('services.sync.numClients');
+  client = yield driver.client();
+  assert.equal(client.syncMobileDevices, 0);
+  assert.equal(client.syncDesktopDevices, 0);
+  assert.equal(client.syncTotalDevices, 0);
+}));
+
 before(exports, () => {
   sandbox = new Cu.Sandbox(null);
-  driver = new NormandyDriver(sandbox, {});
+  // Passing a fake reciperRunner for now; this should either be mocked
+  // properly or removed as an argument to the driver.
+  driver = new NormandyDriver({}, sandbox, {});
```

Comment:
I don't like leaving this comment here, but it's true and I'm not a fan of leaving the opaque objects unexplained.